### DISCUSSION TITLE
Out of the box JWT Authentication?

### DISCUSSION THREAD

**Discussion by xmarkclx** [adonisjs/core#2166] [General]
Is there an official method of using JWT for auth?

If not, do you think it is fine to use the current OAT one over JWT, securitywise?

**Comment by thetutlage** [comment]
Security wise OAT are far superior that JWT. In fact, JWT is something I recommend everyone to not use at all, unless you have a very specific use case that only JWT tokens can solve.

I recommend reading following resources to understand why

- http://cryto.net/~joepie91/blog/2016/06/13/stop-using-jwt-for-sessions/
- http://cryto.net/%7Ejoepie91/blog/2016/06/19/stop-using-jwt-for-sessions-part-2-why-your-solution-doesnt-work/
- https://github.com/adonisjs/core/discussions/2039#discussioncomment-201869

**Comment by alex-oliveira** [comment]
I did an implementation using adonis guard, which I consider to be the "most correct" way.

https://github.com/adonisjs/core/discussions/2506#discussioncomment-825904
